### PR TITLE
Adding functions for updating or deleting tasks

### PR DESCRIPTION
**Pull request by KyleOrth**

### CODE REVIEW

**Comment by Morasiu** [comment]
Great PR! I cannot believe that I've missed those features!

Just add tests for it and I'll more than happy to merge it! Thanks @KyleOrth

**Comment by KyleOrth** [comment]
@Morasiu I have added in some tests for these new functions, hopefully it is all good there - let me know if you have any other comments.

File: Clockify.Tests/Tests/TaskTests.cs [hunk 1]
```diff
@@ -86,5 +86,69 @@ public async Task CreateTaskAsync_NullName_ShouldThrowArgumentException()
             await create.Should().ThrowAsync<ArgumentException>()
                 .WithMessage($"Value cannot be null. (Parameter '{nameof(TaskRequest.Name)}')");
         }
+
+        [Test]
+        public async Task UpdateTaskAsync_ShouldUpdateTask()
+        {
+            // Prepare some project
+            var projectRequest = new ProjectRequest { Name = "Update task project", Color = "#FF00FF" };
```
Comment (Morasiu) [inline Clockify.Tests/Tests/TaskTests.cs:94]: Change:
`var projectRequest = new ProjectRequest { Name = "Update task project", Color = "#FF00FF" };`
  
To this 
`var projectRequest = new ProjectRequest { Name = "Update task project" + Guid.NewGuid()`, Color = "#FF00FF" };`

File: Clockify.Tests/Tests/TaskTests.cs [hunk 2]
```diff
@@ -86,5 +86,69 @@ public async Task CreateTaskAsync_NullName_ShouldThrowArgumentException()
...
+            var findResult2 = await _client.FindAllTasksAsync(_workspaceId, projectId);
+            findResult2.IsSuccessful.Should().BeTrue();
+            findResult2.Data.Should().ContainEquivalentOf(updateResult.Data);
+        }
+
+        [Test]
+        public async Task DeleteTaskAsync_ShouldRemoveTask()
+        {
+            // Prepare some project
+            var projectRequest = new ProjectRequest { Name = "Delete task project", Color = "#FF00FF" };
```
Comment (Morasiu) [inline Clockify.Tests/Tests/TaskTests.cs:129]: Change:
```
var projectRequest = new ProjectRequest { Name = "Delete task project", Color = "#FF00FF" };
```

To this:
```
 var projectRequest = new ProjectRequest { Name = "Delete task project" + Guid.NewGuid(), Color = "#FF00FF" };
```

**Comment by Morasiu** [comment]
Almost done :)

But I've noticed that you are trying to merge these changes to branch `master` instead of `develop`.  I've changed it for you so don't worry  👍

**Comment by KyleOrth** [comment]
@Morasiu Thanks for updating the branch to be correct. I have made the test changes requested  👍

**Pull request merged**